### DISCUSSION TITLE
Please add an option to force the default language for Whisper.

### DISCUSSION THREAD

**Discussion by Artur4learning** [open-webui/open-webui#2025] [Ideas]
I use Whisper often, but unfortunately sometimes instead of English,
the text is entered in Chinese or some other random language.
Would it be possible to add an option to force the typed language?
Regards

**Comment by traklo** [comment]
I'd like to contribute to this but not entirely sure how to start, could someone read through the steps I can see after sifting through the codebase?

Frontend:
1. Add language selection option to Admin's [STT Settings](https://github.com/open-webui/open-webui/blob/main/src/lib/components/admin/Settings/Audio.svelte#L179) if Whisper is selected? This would be detected in [individual user settings](https://github.com/open-webui/open-webui/blob/main/src/lib/components/chat/Settings/Audio.svelte#L99) depending on default set by Admin.

Backend:
2. Add a field in config for [AUDIO_STT_MODEL](https://github.com/open-webui/open-webui/blob/main/backend/open_webui/config.py#L1446) which would have default as `detect` unless the admin/user sets a specific language?

3. Include a new field in [audio variables setting](https://github.com/open-webui/open-webui/blob/main/backend/open_webui/apps/audio/main.py#L55).

5. Apply selected language in [the call to transcribe](https://github.com/open-webui/open-webui/blob/main/backend/open_webui/apps/audio/main.py#L357) if it exists (akin to what's proposed [in this thread](https://github.com/openai/whisper/discussions/1185#discussioncomment-5501213)), otherwise detect (as is)?

Could this be a good first issue for me to do? I would appreciate your guidance.

**Comment by Thaniel94** [comment]
(*Edit* This change is now on the dev branch)

Openai calls Whisper's transcribe function in audio.py. (https://github.com/open-webui/open-webui/blob/main/backend/open_webui/routers/audio.py)
It's possible to add a 'language' argument in there to lock the language to a specific option (e.g. 'en' for English).


Maybe I missed the setting, but going in and editing audio.py to set the argument explicitly fixed the issue in my case, but it would be great if someone could add in  an environmental variable to sit alongside the whisper ENV variables (e.g. WHISPER_MODEL, WHISPER_MODEL_DIR). Maybe calling it WHISPER_LANGUAGE, and having it override the whisper language when set - seems like it would be a quick change, while also providing a lot of value and opportunity for future improvement (e.g. adding the ui option to force language on the frontend later) - though that's assuming the only place where the change would be needed would be in this area with 'model.transcribe' (I don't see why not though - this worked in my manual edit).


I also found I could not lock the default language of whisper, and since the language I wanted was not my first, I found it was having trouble figuring out the right language to use during STT. So if this could be added it would be great (unless there's already a way of doing this I missed):




In [audio.py](https://github.com/open-webui/open-webui/blob/main/backend/open_webui/routers/audio.py):

segments, info = model.transcribe(
            file_path,
            beam_size=5,
            vad_filter=request.app.state.config.WHISPER_VAD_FILTER,
           language = 'en' # - turn this into an env variable - WHISPER_LANGUAGE (Don't set to begin with)
        )

**Reply by Thaniel94** [reply]
Apologies if anyone's already made a start on this, I should have mentioned it earlier, but just confirming I decided to make the small changes myself, and the 'WHISPER_LANGUAGE' variable has been approved and added to the dev branch. It should work as described when using this branch, and with luck may be added to the stable branch when the time comes:
https://github.com/open-webui/open-webui/pull/13376

As suggested by traklo, possible further enhancement would be an ability to configure it on the frontend in the appropriate place, perhaps as a dropdown with all the possible languages (Consider changing it to a persistent config variable if so).
